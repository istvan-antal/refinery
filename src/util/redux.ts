
import { Action } from 'redux';

export interface ActionWithData<T extends string, P> extends Action<T> {
    payload: P;
}

/* eslint import/export: 0 */

export function createAction<T extends string>(type: T): Action<T>;
export function createAction<T extends string, P>(type: T, payload: P): ActionWithData<T, P>;
export function createAction<T extends string, P>(type: T, payload?: P) {
    return payload === undefined ? { type } : ({ type, payload });
}

// tslint:disable-next-line:no-any
type FunctionType = (...args: any[]) => any;
interface ActionCreatorMapObject { [actionCreator: string]: FunctionType }
export type ActionsUnion<A extends ActionCreatorMapObject> = ReturnType<A[keyof A]>;