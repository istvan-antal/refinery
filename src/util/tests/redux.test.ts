import { createAction } from '../redux';

test('createAction with parameter', () => {
    expect(createAction('test', 'test')).toMatchSnapshot();
});

test('createAction without parameter', () => {
    expect(createAction('test')).toMatchSnapshot();
});