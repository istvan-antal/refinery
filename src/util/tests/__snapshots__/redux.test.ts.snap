// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createAction with parameter 1`] = `
Object {
  "payload": "test",
  "type": "test",
}
`;

exports[`createAction without parameter 1`] = `
Object {
  "type": "test",
}
`;
